pdns/dnsdist-rings.cc

```diff
@@ -24,6 +24,49 @@
 
 #include "dnsdist-rings.hh"
 
+void Rings::setCapacity(size_t newCapacity, size_t numberOfShards)
+{
+  if (d_initialized) {
+    throw std::runtime_error("Rings::setCapacity() should not be called once the rings have been initialized");
+  }
+  d_capacity = newCapacity;
+  d_numberOfShards = numberOfShards;
+}
+
+void Rings::init()
```

Comment:
There's is a race if two threads calls `init()` on the same object, but I suppose init() is called when still in single thread mode?
The race is e.g. on `d_shars.resize(...)`

Reply:
Yes, it should be called only once, in the main thread. I could use an `atomic_flag` if you think it makes things more clear.

Reply:
No hard preference, but it might avoid mistakes in the future when this code is used in another context.